homework9.md

```diff
@@ -0,0 +1,207 @@
+23:35:59.699038 IP (tos 0x2,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto TCP (6), length 136)
+    192.168.56.1.60106 > localhost.localdomain.ssh: Flags [P.], cksum 0xb055 (correct), seq 1589:1673, ack 1794, win 2048, options [nop,nop,TS val 646547702 ecr 2858602], length 84
+23:35:59.730333 IP (tos 0x0, ttl 64, id 0, offset 0, flags [DF], proto TCP (6), length 52)
+    192.168.56.1.60106 > localhost.localdomain.ssh: Flags [.], cksum 0x9283 (correct), seq 1673, ack 1822, win 2047, options [nop,nop,TS val 646547732 ecr 2860822], length 0
+23:35:59.730537 IP (tos 0x2,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto TCP (6), length 164)
+    192.168.56.1.60106 > localhost.localdomain.ssh: Flags [P.], cksum 0x4a05 (correct), seq 1673:1785, ack 1822, win 2048, options [nop,nop,TS val 646547732 ecr 2860822], length 112
+23:35:59.934062 IP (tos 0x0, ttl 64, id 0, offset 0, flags [DF], proto TCP (6), length 52)
+    192.168.56.1.60106 > localhost.localdomain.ssh: Flags [.], cksum 0x8e91 (correct), seq 1785, ack 2322, win 2040, options [nop,nop,TS val 646547933 ecr 2861026], length 0
+23:35:59.934238 IP (tos 0x0, ttl 64, id 0, offset 0, flags [DF], proto TCP (6), length 52)
+    192.168.56.1.60106 > localhost.localdomain.ssh: Flags [.], cksum 0x8e5e (correct), seq 1785, ack 2366, win 2047, options [nop,nop,TS val 646547933 ecr 2861026], length 0
+23:35:59.934450 IP (tos 0x4a,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto TCP (6), length 496)
+    192.168.56.1.60106 > localhost.localdomain.ssh: Flags [P.], cksum 0x5e76 (correct), seq 1785:2229, ack 2366, win 2048, options [nop,nop,TS val 646547933 ecr 2861026], length 444
+ ```
+#### 3. Close session. Find in tcpdump output packets that are related to TCP session closure.
+```bash
+23:36:17.141533 IP (tos 0x4a,ECT(0), ttl 64, id 0, offset 0, flags [DF], proto TCP (6), length 112)
+    192.168.56.1.60106 > localhost.localdomain.ssh: Flags [P.], cksum 0xa16f (correct), seq 2697:2757, ack 3486, win 2048, options [nop,nop,TS val 646565067 ecr 2878233], length 60
+23:36:17.141552 IP (tos 0x48, ttl 64, id 0, offset 0, flags [DF], proto TCP (6), length 52)
+    192.168.56.1.60106 > localhost.localdomain.ssh: Flags [F.], cksum 0x000b (correct), seq 2757, ack 3486, win 2048, options [nop,nop,TS val 646565067 ecr 2878233], length 0
+23:36:17.151543 IP (tos 0x48, ttl 64, id 0, offset 0, flags [DF], proto TCP (6), length 52)
+    192.168.56.1.60106 > localhost.localdomain.ssh: Flags [.], cksum 0xfff6 (correct), seq 2758, ack 3487, win 2048, options [nop,nop,TS val 646565076 ecr 2878243], length 0
+  ```
+#### 4. run tcpdump and request any http site in separate session. Find HTTP request and answer packets with ASCII data in it.  Tcpdump command must be as strict as possible to capture only needed packages for this http request.
+
+
+```bash
+ curl http://info.cern.ch
+
+[student@localhost ~]$ sudo tcpdump -A -i enp0s3 src 10.0.2.15 and tcp and port http 
```

Comment:
Will this command allow us to get **answer** packets?

Reply:
I fixed it.

```bash
[student@localhost ~]$ sudo tcpdump -A -i enp0s3 src 10.0.2.15 and port 80 and dst info.cern.ch 
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on enp0s3, link-type EN10MB (Ethernet), capture size 262144 bytes
18:21:11.903894 IP localhost.localdomain.47772 > webafs706.cern.ch.http: Flags [S], seq 2314173199, win 29200, options [mss 1460,sackOK,TS val 1888007 ecr 0,nop,wscale 7], length 0
E..<..@.@...
......l...P..{.......r..a.........
............
18:21:11.976316 IP localhost.localdomain.47772 > webafs706.cern.ch.http: Flags [.], ack 280512002, win 29200, length 0
E..(.	@.@...
......l...P..{...F.P.r..M..
18:21:11.976632 IP localhost.localdomain.47772 > webafs706.cern.ch.http: Flags [P.], seq 0:76, ack 1, win 29200, length 76: HTTP: GET / HTTP/1.1
E..t.
@.@..F
......l...P..{...F.P.r.....GET / HTTP/1.1
User-Agent: curl/7.29.0
Host: info.cern.ch
Accept: */*


18:21:12.050151 IP localhost.localdomain.47772 > webafs706.cern.ch.http: Flags [.], ack 879, win 30730, length 0
E..(..@.@...
......l...P..{\..IpP.x
.M..
18:21:12.050508 IP localhost.localdomain.47772 > webafs706.cern.ch.http: Flags [F.], seq 76, ack 879, win 30730, length 0
E..(..@.@...
......l...P..{\..IpP.x
.M..
18:21:12.050543 IP localhost.localdomain.47772 > webafs706.cern.ch.http: Flags [.], ack 880, win 30730, length 0
E..(..@.@...
......l...P..{]..IqP.x
.M..
^C
6 packets captured
10 packets received by filter
0 packets dropped by kernel

```

Reply:
Ok
Will this command return exactly http packets? Or it will return everything from port 80 (which can be reconfigured another protocol)?

Reply:
It will capture only packets from 80 port.